app/lib/actions/katello/host/update.rb

```diff
@@ -26,7 +26,7 @@ def plan(host, consumer_params = nil)
         def run
           User.as_anonymous_admin do
             host = ::Host.find(input[:host_id])
-            unless input[:consumer_params][:facts].blank?
+            if input[:consumer_params].try(:facts)
```

Comment:
don't you mean  `input[:consumer_params].try(:[], :facts)` ?   Does this actually work?

Reply:
so it actually works in the case I tested, but you are correct, its not the best solution.
This syntax will always return nil:
```
irb(main):006:0* input
=> {:consumer_params=>nil}
irb(main):016:0> input[:consumer_params].try(:facts)
=> nil
irb(main):012:0> input[:consumer_params] = {facts: "hi"}
=> {:facts=>"hi"}
irb(main):015:0> input[:consumer_params].try(:facts)
=> nil
```
I'll update to `.try(:[], :facts)`, good catch

Reply:
apparently you can also do this:
```
irb(main):024:0* input.try(:why).try(:does).try(:this).try(:not).try(:error)
=> nil
```
:joy:

Reply:
Actually this makes sense the more I think about it :smile: